src/persist-client/src/read.rs

```diff
@@ -270,6 +270,25 @@ where
         ret
     }
 
+    /// Test helper to read from the listener until the the shard is closed and
+    /// all events have been emitted.
+    #[cfg(test)]
+    #[track_caller]
```

Comment:
it's supposed to make panics point at the caller. e.g. this is how if you call `.unwrap()` on a None, the panic points at your callsite instead of the literal panic call in option.rs in the stdlib. for some reason (I suspect `async fn`) this doesn't seem to be actually working in practice right now but /shrug

looks like this one doesn't have any panics, I just blindly slapped them on all the cfg(test) helpers in case any of them get cargo culted